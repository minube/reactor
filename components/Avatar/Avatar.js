import React, { Component } from 'react';
import { bool, string } from 'prop-types';
import { StyleSheet, View } from 'react-native';

import Activity from '../Activity';
import styles from './Avatar.style';
import Image from '../Image';
import Link from '../Link';
import { THEME } from '../../common';

const {
   AVATAR,
} = THEME;

const Avatar = ({
  small, medium, big, image, href, style
}) => (
  <View >
    <Link href={href}>
      <Image source={{ uri: image }}
        style={StyleSheet.flatten([
          small && styles.small,
          medium && styles.medium,
          big && styles.big,
          style,
        ])}/>
    </Link>
  </View>
);

Avatar.propTypes = {
  small: bool,
  medium: bool,
  big: bool,
  image: string,
  link:string
};

Avatar.defaultProps = {
  small: false,
  medium: false,
  big: false,
  image: undefined,
  link:undefined
};

export default Avatar;
