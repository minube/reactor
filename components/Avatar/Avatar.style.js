import { StyleSheet } from 'react-native';

import { THEME } from '../../common';

const { AVATAR: { REGULAR, SMALL, LARGE } } = THEME;

export default StyleSheet.create({
  image: {
    borderRadius: REGULAR / 2,
    height: REGULAR,
    width: REGULAR,
  },

  small: {
    borderRadius: SMALL / 2,
    height: SMALL,
    width: SMALL,
  },

  large: {
    borderRadius: LARGE / 2,
    height: LARGE,
    width: LARGE,
  },
});
