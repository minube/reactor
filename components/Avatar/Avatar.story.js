import React from 'react';
import { storiesOf } from '@storybook/react';
import { boolean } from '@storybook/addon-knobs/react';

import WebView from '../WebView';
import Avatar from './Avatar';
import { THEME } from '../../common';

const IMAGE_AVATAR = 'https://avatars.mnstatic.com/117881_80x80_5.jpg';
const LINK_AVATAR = 'https://www.minube.com';
const RADIUS_AVATAR = 100;
const {
   AVATAR,
} = THEME;

storiesOf('🛠 Avatar', module)
  .addWithJSX('default', () => (
    <Avatar regular resizeMode="cover" style={{ borderRadius:'50%' }} image={IMAGE_AVATAR} href={LINK_AVATAR} />
  ))
  .addWithJSX('small', () => (
    <Avatar small resizeMode="cover" style={{ borderRadius:'50%' }} image={IMAGE_AVATAR} />
  ))
  .addWithJSX('large', () => (
    <Avatar large resizeMode="cover" style={{ borderRadius:'50%' }} image={IMAGE_AVATAR} />
  ))
  .addWithJSX('link', () => (
    <Avatar regular resizeMode="cover" style={{ borderRadius:'50%' }} image={IMAGE_AVATAR} href={LINK_AVATAR} />
  ))
  .addWithJSX('🏀 Playground', () => (
    <WebView>
      <Avatar
        style={{ borderRadius:'50%' }}
        resizeMode="cover"
        image={IMAGE_AVATAR}
        small={boolean('small', false)}
        regular={boolean('regular', true)}
        large={boolean('large', false)}
      />
    </WebView>
  ));
