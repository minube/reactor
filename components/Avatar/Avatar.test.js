import React from 'react';
import renderer from 'react-test-renderer';

import Avatar from './Avatar';

const LOREM_AVATAR = 'https://api.adorable.io/avatars/128/8';
const STYLE = { borderRadius: 5, width: 128, height: 128 };

describe('<Avatar>', () => {
  it('renders', () => {
    const tree = renderer.create(<Avatar image={LOREM_AVATAR} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {small}', () => {
    const tree = renderer.create(<Avatar image={LOREM_AVATAR} small />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {large}', () => {
    const tree = renderer.create(<Avatar image={LOREM_AVATAR} large />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {style}', () => {
    const tree = renderer.create(<Avatar image={LOREM_AVATAR} style={STYLE} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
