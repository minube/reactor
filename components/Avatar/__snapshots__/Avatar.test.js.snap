// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Avatar> renders 1`] = `
<View
  style={
    Array [
      Object {
        "alignContent": "center",
        "alignItems": "center",
        "backgroundColor": "#ffffff",
        "justifyContent": "center",
      },
      Array [
        Object {
          "borderRadius": 20,
          "height": 40,
          "width": 40,
        },
        false,
        false,
        undefined,
      ],
    ]
  }
>
  <Image
    onLoad={[Function]}
    resizeMode="cover"
    source={
      Object {
        "uri": "https://api.adorable.io/avatars/128/8",
      }
    }
    style={
      Array [
        Object {
          "borderRadius": 20,
          "height": 40,
          "width": 40,
        },
        false,
        false,
        undefined,
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#cdd2d4",
          "minHeight": 10,
          "minWidth": 10,
        },
        Object {
          "height": "100%",
          "left": 0,
          "position": "absolute",
          "top": 0,
          "width": "100%",
        },
      ]
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#ffffff",
          "height": "100%",
          "opacity": 0,
          "width": "100%",
        }
      }
    />
  </View>
</View>
`;

exports[`<Avatar> when {large} 1`] = `
<View
  style={
    Array [
      Object {
        "alignContent": "center",
        "alignItems": "center",
        "backgroundColor": "#ffffff",
        "justifyContent": "center",
      },
      Array [
        Object {
          "borderRadius": 20,
          "height": 40,
          "width": 40,
        },
        Object {
          "borderRadius": 30,
          "height": 60,
          "width": 60,
        },
        false,
        undefined,
      ],
    ]
  }
>
  <Image
    onLoad={[Function]}
    resizeMode="cover"
    source={
      Object {
        "uri": "https://api.adorable.io/avatars/128/8",
      }
    }
    style={
      Array [
        Object {
          "borderRadius": 20,
          "height": 40,
          "width": 40,
        },
        Object {
          "borderRadius": 30,
          "height": 60,
          "width": 60,
        },
        false,
        undefined,
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#cdd2d4",
          "minHeight": 10,
          "minWidth": 10,
        },
        Object {
          "height": "100%",
          "left": 0,
          "position": "absolute",
          "top": 0,
          "width": "100%",
        },
      ]
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#ffffff",
          "height": "100%",
          "opacity": 0,
          "width": "100%",
        }
      }
    />
  </View>
</View>
`;

exports[`<Avatar> when {small} 1`] = `
<View
  style={
    Array [
      Object {
        "alignContent": "center",
        "alignItems": "center",
        "backgroundColor": "#ffffff",
        "justifyContent": "center",
      },
      Array [
        Object {
          "borderRadius": 20,
          "height": 40,
          "width": 40,
        },
        false,
        Object {
          "borderRadius": 10,
          "height": 20,
          "width": 20,
        },
        undefined,
      ],
    ]
  }
>
  <Image
    onLoad={[Function]}
    resizeMode="cover"
    source={
      Object {
        "uri": "https://api.adorable.io/avatars/128/8",
      }
    }
    style={
      Array [
        Object {
          "borderRadius": 20,
          "height": 40,
          "width": 40,
        },
        false,
        Object {
          "borderRadius": 10,
          "height": 20,
          "width": 20,
        },
        undefined,
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#cdd2d4",
          "minHeight": 10,
          "minWidth": 10,
        },
        Object {
          "height": "100%",
          "left": 0,
          "position": "absolute",
          "top": 0,
          "width": "100%",
        },
      ]
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#ffffff",
          "height": "100%",
          "opacity": 0,
          "width": "100%",
        }
      }
    />
  </View>
</View>
`;

exports[`<Avatar> when {style} 1`] = `
<View
  style={
    Array [
      Object {
        "alignContent": "center",
        "alignItems": "center",
        "backgroundColor": "#ffffff",
        "justifyContent": "center",
      },
      Array [
        Object {
          "borderRadius": 20,
          "height": 40,
          "width": 40,
        },
        false,
        false,
        Object {
          "borderRadius": 5,
          "height": 128,
          "width": 128,
        },
      ],
    ]
  }
>
  <Image
    onLoad={[Function]}
    resizeMode="cover"
    source={
      Object {
        "uri": "https://api.adorable.io/avatars/128/8",
      }
    }
    style={
      Array [
        Object {
          "borderRadius": 20,
          "height": 40,
          "width": 40,
        },
        false,
        false,
        Object {
          "borderRadius": 5,
          "height": 128,
          "width": 128,
        },
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#cdd2d4",
          "minHeight": 10,
          "minWidth": 10,
        },
        Object {
          "height": "100%",
          "left": 0,
          "position": "absolute",
          "top": 0,
          "width": "100%",
        },
      ]
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#ffffff",
          "height": "100%",
          "opacity": 0,
          "width": "100%",
        }
      }
    />
  </View>
</View>
`;
